src/cascadia/TerminalApp/Profile.cpp

```diff
@@ -368,16 +316,17 @@ bool Profile::ShouldBeLayered(const Json::Value& json) const
         return false;
     }
 
-    const auto& otherSource = json.isMember(JsonKey(SourceKey)) ? json[JsonKey(SourceKey)] : Json::Value::null;
+    std::optional<std::wstring> otherSource;
+    bool otherHadSource = JsonUtils::GetValueForKey(json, SourceKey, otherSource);

```

Comment:
I just read over this and was like, "I guess i made it return booleans for a reason. cool!"

(this is so that we treat `source:null` as having a source, _not_ as _not having a source_) (GVFK for null returns true, but the optional would be nullopt)